src/backend/marsha/core/lti.py

```diff
@@ -84,7 +85,17 @@ def verify(self):
         ):
             raise LTIException()
 
-        return passport.consumer_site or passport.playlist.consumer_site
+        consumer_site = passport.consumer_site or passport.playlist.consumer_site
+
+        # Make sure we only accept requests from a domain matching the consumer_site associated
+        # with the passport.
+        if (
+            urlparse(self.request.META.get("HTTP_REFERER")).hostname
```

Comment:
I'm not sure we can trust the `referer` header. This header is added by the browser so it can be craft by an attacker. the [Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) header is maybe the one to use ans you should also take care of `X-Forwarded-For` if we are behind a proxy (like in a k8s environment). Maybe Django has helper to retrieve the trusted host, it should be better than digging ourself in the request headers.

Reply:
django has the perfect method you need : https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpRequest.get_host

Reply:
I don't think host is what we need here.

The HOST header for the iframe is Marsha's domain (or at least I think it is? That's what I saw in my testing). What we're looking for is the domain for the page that includes the iframe.

Reply:
indeed `host` is not the good one but `referer` is still not what we can trust. Maybe we should also look Content Security Policy added in our response https://content-security-policy.com/

Reply:
> This header is added by the browser so it can be craft by an attacker.

What we are trying to prevent here is: "someone we trust and gave a passport to is using the passport on another site than the site for which we gave them the passport"

I don't think our ambition here is to block an attacker who would handcraft a request after stealing valid passport credentials...

Reply:
well no matter the case we are, the first rule you learn is **never ever trust client**. Referer is set by the client so this value should not be used.

CSP is a very good alternative IMO, well configured we can prevent some XSS. 
But it doesn't resolve the problem we have, whitelisting a trusted domain.

Reply:
This is not a case of trusting the client but of ensuring our passport is not used on other origins.

AFAIK the referer header **cannot be changed** in the browser.

Reply:
everything can be change in a browser ;-) http://kb.mozillazine.org/Network.http.sendRefererHeader

Reply:
The point of this is not to foil an attacker's browser but to prevent passport reuse... Which would require modifying such settings on the browsers of end users.

Let's talk about this on video :)